src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -0,0 +1,165 @@
+
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                _dte = await asyncServiceProvider.GetDTEAsync();
+                _buildEvents = _dte.Events.BuildEvents;
+                _buildEvents.OnBuildBegin += (_, __) => { ErrorListTableDataSource.Value.ClearNuGetEntries(); };
+                _solutionEvents = _dte.Events.SolutionEvents;
+                _solutionEvents.AfterClosing += () => { ErrorListTableDataSource.Value.ClearNuGetEntries(); };
+                OutputConsole = await consoleProvider.CreatePackageManagerConsoleAsync();
+            });
+        }
+
+        public void Dispose()
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                ErrorListTableDataSource.Value.Dispose();
+            });
+        }
+
+        public void End()
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await OutputConsole.WriteLineAsync(Resources.Finished);
+                await OutputConsole.WriteLineAsync(string.Empty);
+
+                // Give the error list focus
+                ErrorListTableDataSource.Value.BringToFrontIfSettingsPermit();
+            });
+        }
+
+        public void Log(MessageLevel level, string message, params object[] args)
+        {
+            if (level == MessageLevel.Info
+                || level == MessageLevel.Error
+                || level == MessageLevel.Warning
+                || _verbosityLevel > DefaultVerbosityLevel)
+            {
+                if (args.Length > 0)
+                {
+                    message = string.Format(CultureInfo.CurrentCulture, message, args);
+                }
+
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(() => OutputConsole.WriteLineAsync(message));
+            }
+        }
+
+        public void Log(ILogMessage message)
+        {
+            if (message.Level == LogLevel.Information
+                || message.Level == LogLevel.Error
+                || message.Level == LogLevel.Warning
+                || _verbosityLevel > DefaultVerbosityLevel)
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(() => OutputConsole.WriteLineAsync(message.FormatWithCode()));
+
+                if (message.Level == LogLevel.Error ||
+                    message.Level == LogLevel.Warning)
+                {
+                    ReportError(message);
+                }
+            }
+        }
+
+        private async Task<int> GetMSBuildVerbosityLevelAsync()
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var properties = _dte.get_Properties(DTEEnvironmentCategory, DTEProjectPage);
```

Comment:
If this throws an exception, do we need to return `DefaultVerbosityLevel`?

Reply:
i've only moved these types and changed their namespace (but now that you mention it...i don't see the old files being deleted...). I'm not aware of any new requirements on these types yet.
The old code paths (all that is in this PR), all work fine as is.

As i bring in changes to N.VS.OE.Client that use these same types, we may need to modify them to not rely on DTE, or other changes.

Reply:
deleted those 7 files that i had moved (confirming quality with a local build now). i had removed them from the project, but not the files.
talked with @dominoFire offline about reviewing just the type move.